File: frontend_mobile/lib/data/models/recipe/trending_tree_dto.dart
import 'package:json_annotation/json_annotation.dart';

part 'trending_tree_dto.g.dart';

@JsonSerializable()
class TrendingTreeDto {
  final String rootRecipeId;
  final String title;
  final String? foodName;
  final String cookingStyle;
  final String? thumbnail;
  final int variantCount;
  final int logCount;
  final String? latestChangeSummary;
  final String? userName;
  final String? creatorPublicId;

  TrendingTreeDto({
    required this.rootRecipeId,
    required this.title,
    this.foodName,
    required this.cookingStyle,
    this.thumbnail,
    required this.variantCount,
    required this.logCount,
    this.latestChangeSummary,
    this.userName,
    this.creatorPublicId,
  });

  factory TrendingTreeDto.fromJson(Map<String, dynamic> json) =>
      _$TrendingTreeDtoFromJson(json);
  Map<String, dynamic> toJson() => _$TrendingTreeDtoToJson(this);
}

File: backend/src/main/java/com/pairingplanet/pairing_planet/dto/recipe/TrendingTreeDto.java
package com.pairingplanet.pairing_planet.dto.recipe;

import lombok.Builder;

import java.util.UUID;

@Builder
public record TrendingTreeDto(
        UUID rootRecipeId,
        String title,
        String foodName,      // 음식 이름
        String culinaryLocale,
        String thumbnail,     // 썸네일 URL
        Long variantCount,    // 변형 수
        Long logCount,        // 로그 수
        String latestChangeSummary, // 최근 변형 사유 요약
        String userName,   // 작성자 이름
        UUID creatorPublicId  // 작성자 publicId (프로필 링크용)
) {}